plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Evaluation/Machine/Cek/Internal.hs

```diff
@@ -425,17 +437,26 @@ type Context uni fun = [Frame uni fun]
 runCekM
     :: forall a cost uni fun.
     (PrettyUni uni fun)
-    => BuiltinsRuntime fun (CekValue uni fun)
+    => MachineParameters CekMachineCosts CekValue uni fun
     -> ExBudgetMode cost uni fun
     -> Bool
     -> (forall s. (GivenCekReqs uni fun s) => CekM s a)
     -> (Either (CekEvaluationException uni fun) a, cost, [String])
-runCekM runtime (ExBudgetMode getExBudgetInfo) emitting a = runST $ do
+runCekM (MachineParameters costs runtime) (ExBudgetMode getExBudgetInfo) emitting a = runST $ do
     exBudgetMode <- getExBudgetInfo
     mayLogsRef <- if emitting then Just <$> newSTRef DList.empty else pure Nothing
     let ?cekRuntime = runtime
         ?cekEmitter = mayLogsRef
         ?cekBudgetSpender = _exBudgetModeSpender exBudgetMode
+        ?cekCosts = costs
+        -- We have the slippage in terms of *budget*, this computes the slippage in terms of *steps*, which
+        -- is what the machine wants.
+        --
+        -- Taking the minimum with 1000 avoids us picking a ridiculous number of steps if we're using
+        -- the enormous budget.
+        -- TODO: make the benchmarks run things with the minimum required budget instead
+        -- so we won't need this.
+        ?cekSlippage = min 1000 (fromIntegral $ _exBudgetModeSlippage exBudgetMode `divBudget` cekStepCost costs)
```

Comment:
I found it quite difficult to understand what was going on here because it's tricky to work out where the slippage number is coming from.  If I understand correctly we're only adding up the machine step costs ten times per program (or once every 1000 steps for big programs).   I don't think this is actually stated explicitly anywhere though: that would have made things a lot easier to understand.